lib/util.js

```diff
@@ -22,6 +22,13 @@ const fs = require('fs');
  */
 function createReadStream(path, options) {
     return new Promise((resolve, reject) => {
+        if((typeof options === 'string' || options instanceof String) || 
+            (typeof options === 'object')) { // ensure flags are here                
+            options.flags = options.flags || 'r';
+            options.encoding = options.encoding || 'utf8';
+            options.mode = options.mode || 0o444;
```

Comment:
If `typeof options === 'string'` then we don't want to set properties on it (e.g. `options.flags = ...`), right?

Was this added to try and fix it for Windows? If yes, and the change is slightly speculative and it overall doesn't work on Windows yet, I would revert it to the known good state from before to avoid any unwanted regressions.

Reply:
if the type is string, it's only the encoding (utf-8). But right, I need to formulate that more clearly.

Reply:
Nevermind. Windows is fixed. I improved this part of the code in the process.